tests/unit/blocks/blockRegistration.test.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * @jest-environment jsdom
+ */
+import 'jest-expect-message'
+import { buildWindowObject, getAttributes } from '../../../scripts/editor/register-blocks';
+import wrapperManifest from '../../../blocks/init/src/Blocks/wrapper/manifest.json';
+import globalManifest from '../../../blocks/init/src/Blocks/manifest.json';
+import { blockAttributes } from '../../data/block-attributes';
+import { expect } from '@jest/globals';
+import { getAllComponentManifests, getAllBlockManifests, getComponentDependencies, recursiveBuildProps } from '../../helpers/blocks';
+import { propsOutput } from '../../data/props-output';
+
+/**
+ * Returns the name of the component from builtProps.
+ *
+ * @param {object} builtProps Built props for a component
+ * @returns {string}
+ */
+const debugName = (builtProps) => {
+	return `"${builtProps.realName}" (newName: "${builtProps.newName || builtProps.realName}") component`;
+}
+
+/**
+ * Asserts that built props match the expected output.
+ *
+ * @param {object} expectedProps propsOutput object for specific component
+ * @param {object} builtProps Props built for this object.
+ */
+const recursiveAssertProps = (expectedProps, builtProps) => {
+
+	// Some sanity checkcs
+	expect(
+		builtProps,
+		`Missing attributes key in built props for: ${debugName(builtProps)}`
+	).toHaveProperty('attributes');
+
+	// Check expected
+	for (const expected of expectedProps.expected) {
+		expect(
+			builtProps.attributes,
+			`Missing expected property in built props for: ${debugName(builtProps)}`
+		).toHaveProperty(expected);
+	}
+
+	// Check not expected
+	for (const notExpected of expectedProps.notExpected) {
+		expect(builtProps.attributes).not.toHaveProperty(notExpected);
+	}
+
+	// Check prefix
+	expect(
+		builtProps.attributes,
+		`Missing "prefix" key in built props for ${debugName(builtProps)}`
+	).toHaveProperty('prefix');
+	expect(
+		builtProps.attributes.prefix,
+		`Prefix key not correct when sending props to ${debugName(builtProps)}`
+	).toEqual(expectedProps.prefix);
+
+	// Repeat if expected has subComponents
+	if (expectedProps.components && Object.keys(expectedProps.components).length > 0) {
+		for (const componentKey of Object.keys(expectedProps.components)) {
+
+			// Expect built props has subComponents
+			expect(
+				buildProps,
+				`Missing "subComponents" key in built props for ${debugName(builtProps)}. Expecting one because this has subComponents defined in props-output.js`
+			).toHaveProperty('subComponents');
+			expect(
```

Comment:
```suggestion

			expect(
```

Just add space between these two, makes it more readable.